addons/website_sale/controllers/main.py

```diff
@@ -1711,7 +1711,7 @@ def _get_express_shop_payment_values(self, order, **kwargs):
             'transaction_route': f'/shop/payment/transaction/{order.id}',
             'express_checkout_route': self._express_checkout_route,
             'landing_route': '/shop/payment/validate',
-            'payment_method_unknown_id': request.env.ref('payment.payment_method_unknown').id,
+            'payment_method_unknown_id': (payment_method := request.env.ref('payment.payment_method_unknown', raise_if_not_found=False)) and payment_method.id,
```

Comment:
This makes the line quite long and difficult to read; it would be better to extract `payment_method` into a variable first.